x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameIndexerTransformStats.java

```diff
@@ -102,23 +147,49 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.field(SEARCH_TIME_IN_MS.getPreferredName(), searchTime);
         builder.field(SEARCH_TOTAL.getPreferredName(), searchTotal);
         builder.field(SEARCH_FAILURES.getPreferredName(), searchFailures);
+        builder.field(EXPONENTIAL_AVG_CHECKPOINT_DURATION_MS.getPreferredName(), this.expAvgCheckpointDurationMs);
+        builder.field(EXPONENTIAL_AVG_DOCUMENTS_INDEXED.getPreferredName(), this.expAvgDocumentsIndexed);
+        builder.field(EXPONENTIAL_AVG_DOCUMENTS_PROCESSED.getPreferredName(), this.expAvgDocumentsProcessed);
+        builder.field(CONTINUOUS_CHECKPOINTS_PROCESSED.getPreferredName(), continuousCheckpointsProcessed);
         builder.endObject();
         return builder;
     }
 
-    public DataFrameIndexerTransformStats merge(DataFrameIndexerTransformStats other) {
-        numPages += other.numPages;
-        numInputDocuments += other.numInputDocuments;
-        numOuputDocuments += other.numOuputDocuments;
-        numInvocations += other.numInvocations;
-        indexTime += other.indexTime;
-        searchTime += other.searchTime;
-        indexTotal += other.indexTotal;
-        searchTotal += other.searchTotal;
-        indexFailures += other.indexFailures;
-        searchFailures += other.searchFailures;
+    public double getExpAvgCheckpointDurationMs() {
+        return expAvgCheckpointDurationMs;
+    }
+
+    public double getExpAvgDocumentsIndexed() {
+        return expAvgDocumentsIndexed;
+    }
+
+    public double getExpAvgDocumentsProcessed() {
+        return expAvgDocumentsProcessed;
+    }
+
+    public long getContinuousCheckpointsProcessed() {
+        return continuousCheckpointsProcessed;
+    }
+
+    public void incrementCheckpointExponentialAverages(long checkpointDurationMs, long docsIndexed, long docsProcessed) {
+        assert checkpointDurationMs >= 0;
+        assert docsIndexed >= 0;
+        assert docsProcessed >= 0;
+        continuousCheckpointsProcessed++;
+        if (continuousCheckpointsProcessed == 1) {
+            expAvgCheckpointDurationMs = checkpointDurationMs;
+            expAvgDocumentsIndexed = docsIndexed;
+            expAvgDocumentsProcessed = docsProcessed;
+        } else {
+            double alpha = 2.0/(continuousCheckpointsProcessed+1);
```

Comment:
The more checkpoints processed the smaller `alpha` gets and the less significant the latest value becomes.  Did you consider using a constant for `alpha`?

Reply:
@davidkyle you are absolutely correct. I mixed up my implementations. I was initially doing some fancy windowing, but determined it was no longer necessary and then did not change my calculation to reflect that. 

Will update it for sure.